dialect/sql/schema/migrate.go

```diff
@@ -159,6 +161,19 @@ func (m *Migrate) Create(ctx context.Context, tables ...*Table) error {
 	return creator.Create(ctx, tables...)
 }
 
+// Diff compares the state read from the StateReader with the state defined by Ent.
+// Changes will be written to migration files by the configures Planner.
+func (m *Migrate) Diff(ctx context.Context, tables ...*Table) error {
+	if m.atlas.dir == nil {
+		return errors.New("no migration directory given")
+	}
+	plan, err := m.atDiff(ctx, m, tables...)
+	if err != nil {
+		return err
+	}
+	return migrate.New(nil, m.atlas.dir, m.atlas.fmt).WritePlan(plan)
```

Comment:
May I suggest changing `migrate.New` to `migrate.NewPlanner`?

Reply:
Passing `nil` driver feels odd to me. I don't want to abuse functional options, but consider use something like:
```go
migrate.NewPlanner(
    migrate.PlanDir(...),
    migrate.PlanFormatter(...),
    migrate.PlanDriver(...),
    ...
)
```